tests/test_templatetags.py

```diff
@@ -2392,3 +2392,215 @@ def test_inner_slot_iteration_nested_with_slot_default_and_outer_scope_variable_
+            template: types.django_html = """
+                {% load component_tags %}
+                <div>
+                    {% slot "my_slot" abc=abc 123=var123 %}Default text{% endslot %}
+                </div>
+            """
+
+            def get_context_data(self):
+                return {
+                    "abc": "def",
+                    "var123": 456,
+                }
+
+        template: types.django_html = """
+            {% load component_tags %}
+            {% component "test" %}
+                {% fill "my_slot" slot_data="slot_data_in_fill" as "slot_var" %}
+                    {{ slot_var.default }}
+                    {{ slot_data_in_fill.abc }}
+                    {{ slot_data_in_fill.123 }}
+                {% endfill %}
+            {% endcomponent %}
+        """
+        rendered = Template(template).render(Context())
+        expected = """
+            <div>
+                Default text
+                def
+                456
+            </div>
+        """
+        self.assertHTMLEqual(rendered, expected)
+
+    def test_slot_data_raises_on_slot_data_and_as_same_var(self):
+        @component.register("test")
+        class TestComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div>
+                    {% slot "my_slot" abc=abc 123=var123 %}Default text{% endslot %}
+                </div>
+            """
+
+            def get_context_data(self):
+                return {
+                    "abc": "def",
+                    "var123": 456,
+                }
+
+        template: types.django_html = """
+            {% load component_tags %}
+            {% component "test" %}
+                {% fill "my_slot" slot_data="slot_var" as "slot_var" %}
+                    {{ slot_var.default }}
+                {% endfill %}
+            {% endcomponent %}
+        """
+        with self.assertRaisesMessage(
+            TemplateSyntaxError,
+            "'fill' received the same string for slot alias (as ...) and slot data (slot_data=...)",
+        ):
+            Template(template).render(Context())
+
+    def test_slot_data_fill_without_data(self):
+        @component.register("test")
+        class TestComponent(component.Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <div>
+                    {% slot "my_slot" abc=abc 123=var123 %}Default text{% endslot %}
+                </div>
+            """
+
+            def get_context_data(self):
+                return {
+                    "abc": "def",
+                    "var123": 456,
+                }
+
+        template: types.django_html = """
+            {% load component_tags %}
+            {% component "test" %}
+                {% fill "my_slot" %}
+                    overriden
+                    {{ slot_data_in_fill }}
```

Comment:
Not sure what this test is testing, as this is just an undefined variable?

Reply:
The undefined variable is not necessary, but the point of this test is that things still work even if user doesn't provide the `slot_data` kwarg

Reply:
I removed the variable in https://github.com/EmilStenstrom/django-components/pull/495/commits/40c2b1af9b6878a922cd2456b76dbd69ba1e0fef